client/my-sites/pages/page/index.js

```diff
@@ -421,42 +410,144 @@ class Page extends Component {
 						siteUrl={ this.props.multisite && this.getSiteDomain() }
 					/>
 				</div>
-				{ ellipsisGridicon }
-				{ popoverMenu }
-				<ReactCSSTransitionGroup
-					transitionName="updated-trans"
-					transitionEnterTimeout={ 300 }
-					transitionLeaveTimeout={ 300 }
-				>
-					{ this.props.buildUpdateTemplate() }
-				</ReactCSSTransitionGroup>
+				{ ellipsisMenu }
+				{ shadowNotice }
 			</CompactCard>
 		);
 	}
 
+	async onShadow( status ) {
+		await this.props.onShadow( this.props.page.global_ID, !! status );
+		return await new Promise( resolve => this.setState( { shadowStatus: status }, resolve ) );
+	}
+
+	async performUpdate( { action, progressNotice, successNotice, errorNotice } ) {
+		await this.onShadow( progressNotice );
+		try {
+			await action();
+			await this.onShadow( successNotice );
+		} catch ( error ) {
+			await this.onShadow( errorNotice );
+		}
+		await sleep( 5000 );
+		await this.onShadow( false );
+	}
+
+	async updatePostStatus( status ) {
+		const { page, translate } = this.props;
+
+		switch ( status ) {
+			case 'delete':
+				const deleteWarning = translate( 'Delete this page permanently?' );
+
+				if ( typeof window === 'object' && window.confirm( deleteWarning ) ) {
+					await this.performUpdate( {
+						action: () => this.props.deletePost( page.site_ID, page.ID ),
+						progressNotice: {
+							status: 'is-error',
+							icon: 'trash',
+							text: translate( 'Deleting…' ),
+						},
+						successNotice: {
+							status: 'is-success',
+							text: translate( 'Page deleted.' ),
+						},
+						errorNotice: {
+							status: 'is-error',
+							text: translate( 'Failed to delete page.' ),
+						},
+					} );
+				}
+				return;
+
+			case 'trash':
+				await this.performUpdate( {
+					action: () => this.props.trashPost( page.site_ID, page.ID ),
+					progressNotice: {
+						status: 'is-error',
+						icon: 'trash',
+						text: translate( 'Trashing…' ),
+					},
+					successNotice: {
+						status: 'is-success',
+						text: translate( 'Page trashed.' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **4** times:
`translate( 'Page trashed' )` *ES Score: 16*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).